src/conversions/sample_rate.rs

```diff
@@ -50,10 +50,11 @@ where
         let from = from.0;
         let to = to.0;
 
+        assert!(num_channels >= 1);
         assert!(from >= 1);
         assert!(to >= 1);
```

Comment:
Sample rate converter allocates a buffer that is proportional to number of channels (65k * sample size).
Also `SampleRateConverter::size_hint` does not seem to be accurate and it is used to collect items into Vecs in test.
There's a TODO note in it BTW. It seems to be rather complicated, maybe there can be a simpler but imprecise version.